tools/ARIAtools/util/run_logging.py

```diff
@@ -0,0 +1,118 @@
+
+import os
+import pickle
+import json
+from datetime import datetime
+import shapely
+
+import ARIAtools
+import ARIAtools.util.shp
+
+class RunLog:
+    """
+    """
+    def __init__(self, workdir, verbose=None):
+        """
+        """
+        # Record parameters
+        self.workdir = os.path.abspath(workdir)
+
+        self.verbose = verbose
+
+        # Automatically establish log directory and files
+        self.log_dir = os.path.join(self.workdir, 'config')
+        os.makedirs(self.log_dir, exist_ok=True)
+
+        self.log_name = os.path.join(self.log_dir, 'PICKLE.pkl')
+        if not os.path.exists(self.log_name):
+            with open(self.log_name, 'wb') as log_file:
+                pickle.dump({}, log_file)
+
+        self.config_name = os.path.join(self.log_dir, 'config.json')
+        if not os.path.exists(self.config_name):
+            with open(self.config_name, 'w') as config_file:
+                json.dump({}, config_file)
+
+        self.file_list_name = os.path.join(self.log_dir, 'files.json')
+        if not os.path.exists(self.file_list_name):
+            with open(self.file_list_name, 'w') as list_file:
+                json.dump({}, list_file)
+
+        self.extracted_files_name = os.path.join(self.log_dir,
+                                                 'extracted_files.json')
+        if not os.path.exists(self.extracted_files_name):
+            with open(self.extracted_files_name, 'w') as extr_file:
+                json.dump({}, extr_file)
+
+
+    def load(self):
+        """
+        """
+        with open(self.log_name, 'rb') as log_file:
+            log_data = pickle.load(log_file)
```

Comment:
for f in log_data['files']:
    if ~os.path.exists(f):
              print('WARNING, {f} path changed')